### DISCUSSION TITLE
Should this be reasonably simple to make work with Laravel & Inertia ?

### DISCUSSION THREAD

**Discussion by Foxxat** [vueform/vueform#297] [Questions]
I've battled this one all day and not managed to get it working sensibly.

Either vueform submits the form and effectively bypasses inertia, of inertia submits the form and doesn't have the input field values from vueform. I can't decide if the approach is fundamentally incompatible or if I'm just not getting it.

I can provide an example but I thought I'd ask the question before I spent any more time on it.

**Comment by bambamboole** [comment]
I am also trying to solve this currently. Current approach kinda works, but is rather hacky:
I am creating an inertia form on the fly and add all the properties. Here is a simple example:
```js
<script setup lang="ts">
import { useForm } from '@inertiajs/vue3';
import { FormSchema } from '@/Components/ui/form/index';
import { HTMLAttributes, ref } from 'vue';
import { cn } from '@/lib/utils';

const props = defineProps<{
    class?: HTMLAttributes['class'];
    schema: FormSchema;
}>();
const vueFormSchema = ref({
    name: { type: 'text', label: 'Name' },
    register: {
        type: 'button',
        submits: true,
        buttonLabel: 'Create Project',
        full: true,
        size: 'lg'
    }
});

const vueForm = ref(null);

const submitForm = async (FormData: FormData, form$: any) => {
    let data: any = {};
    FormData.forEach((value, key) => {
        data[key] = value;
    });

    const inertiaForm = useForm(data);
    inertiaForm.submit(props.schema.method, props.schema.endpoint, {
        preserveScroll: true,
        onError: (errors: object) => {
            console.log(errors);
            Object.entries(errors).forEach(([key, value]) => {
                form$.el$(key).messageBag.prepend(value);
            });
        }
    });
};

</script>

<template>
    <Vueform :ref="vueForm" :endpoint="submitForm" :schema="vueFormSchema" :class="cn('pt-4', props.class)" />
</template>
```
The `schema` is here is not relevant. its coming form the backend and holds form infos, but the vueform schema is still hard coded and I only use method and endpoint from the backend.

My current issue with this approach is, that the error message is shown twice and I have no clue why:
<img width="1347" alt="Screenshot 2024-08-27 at 15 47 32" src="https://github.com/user-attachments/assets/d0b7ba8c-83c2-44c6-9d90-1d237b08319d">


@adamberecz do you maybe have a hint on how to implement The form handling properly with Inertia?
Having this as first class support would be a game changer.

**Comment by bambamboole** [comment]
Ok, I was able to progress a bit, but it still feels very hacky. Since I am using this in a private project, I set up a public repo with a simple implementation: https://github.com/bambamboole/laravel-vueform-example.
I added some points to the readme, for what I want to solve (and what not). 

I am happy for any feedback. Also happy to pair up to tackle the inertia integration issue.
If we can get this into a (better) usable state, I am happy to wire up a package which does all the heavy lifting regarding form element classes and serializing them properly to JSON

**Reply by adamberecz** [reply]
Hey, thanks for taking the initiative on this! I really like the idea of making the project Inertia-compatible, even though I haven't had much time to dive into it myself. I'm not familiar with Inertia, but I'm open to collaborating.

I’ve submitted a PR to handle displaying success messages, but I could use some clarification on the other goals you’ve outlined. It might be most efficient to set up a call to discuss. You can find a time that works for you [here](https://calendly.com/vueform).

**Reply by bambamboole** [reply]
awesome, thank you. I am going on vacation for a few days and be back next week. I will schedule something for us 👍

**Comment by bambamboole** [comment]
I made some improvements in https://github.com/bambamboole/laravel-vueform-example after some help from @adamberecz .
The reactivity is fixed and I have also built a LinkElement. There I have some issue with a local import of inertias `Link` component and had to register it globally in `app.ts`. Maybe someone has a tip on how to fix this.

So the current todo list looks like this:
* [x] Basic setup with Vueform
* [x] POC Update password form
* [x] Show feedback of successful form submission
* [x] POC Update profile form
* [x] Conditional fields 
* [x] Make a proper inertia Link component which does not suck. -> use custom element -> https://vueform.com/docs/creating-elements
* [ ] Figure out why Link can't be imported in LinkElement.vue. I had now to import it globally in app.ts
* [ ] Reimplement needed [inertias useForm](https://github.com/inertiajs/inertia/blob/master/packages/vue3/src/useForm.ts) logic to be more flexible
* [ ] Filled fields have changed border color
* [ ] Add more types
* [ ] Extract into npm package

**Reply by bambamboole** [reply]
I am currently pretty packed and won't find time to finalise this on my own. Help appreciated ;)
